aws/rust-runtime/aws-config/src/imds/client.rs

```diff
@@ -764,6 +766,21 @@ pub(crate) mod test {
     use std::time::{Duration, UNIX_EPOCH};
     use tracing_test::traced_test;
 
+    macro_rules! assert_full_error_contains {
+        ($err:expr, $contains:expr) => {
+            let err = $err;
+            let message = format!(
+                "{}",
+                aws_smithy_types::error::display::DisplayErrorContext(&err)
+            );
+            assert!(
+                message.contains($contains),
+                "Error message '{message}' didn't contain text '{}'",
+                $contains
+            );
+        };
+    }
```

Comment:
I definitely considered it, but I didn't want to create a whole create just for this macro when there are only a few places it would be used outside of `aws-smithy-http`.

It almost seems like we should replace the `aws-smithy-protocol-test` crate with a general purpose testing crate.